ee/WFLY-1160-Provide_ability_to_configure_a_global_directory_which_puts_the_contents_in_the_classpath.adoc

```diff
@@ -0,0 +1,137 @@
+The 'Jar Resource loader' will make available any class inside of the jar file to the applications.
+
+The jar resources of this module will be created iterating over all jar files in alphabetical order and size descending. The resource root order declaration will govern the class loading order for the classes available from the jar resources of this module.
+
+e.g. Suppose that our global directory path is /lib and it contains two jar files, lib-1.jar and lib-2.jar. The module generated internally by the global-directory operation will be equivalent to declare a module.xml with the following resources:
+
+  <resources>
+    <resource-root path="/lib"/>
+    <resource-root path="/lib/lib-1.jar"/>
+    <resource-root path="/lib/lib-2.jar"/>
+  </resources>
+
+The module created from the shared directory will be loaded as soon as the first application is deployed in the server. That means, if the server is started/restarted and there are no applications deployed, then the shared directory is neither scanned nor the module loaded.
+
+In case of domain mode or distributed environments, it is the user responsibility to make the content of the configured global directory consistent across all the server instances, as well as distribute the jar files that they contain.
+
+Removal of a global-directory could be done using the following operation:
+
+  [standalone@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:remove()
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee/global-directory=my-common-libs:remove()
+
+As any other resources, users can read the current configuration of a global-directory using resource-resource operation:
+
+ [standalone@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:read-resource
+ {
+      "outcome" => "success",
+      "result" => {
+          "path" => "lib",
+          "relative-to" => "jboss.home.dir"
+      }
+ }
+
+ [domain@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:read-resource
+ {
+      "outcome" => "success",
+      "result" => {
+          "path" => "lib",
+          "relative-to" => "jboss.server.data.dir"
+      }
+ }
+
+This RFE will not cover the availability of hot JBoss modules redeployments, that means, modifications of existing jar files as well as adding a new jar or removal an existing one will require a complete server restart, however, changes in plain resources as property files will be available immediately to the applications.
+
+It will be allowed only one global directory configuration.
+
+== Issue Metadata
+
+=== Issue
+* https://issues.jboss.org/browse/EAP7-1124[EAP7-1124]
+* https://issues.jboss.org/browse/WFLY-1160[WFLY-1160]
+* https://issues.jboss.org/browse/WFCORE-4375[WFCORE-4375]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/MODULES-99[MODULES-99]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:vmarek@redhat.com[Vratislav Marek]
```

Comment:
Please add me as QE contact.
`* mailto:tterem@redhat.com[Tomas Terem]`

Reply:
Hello @tterem, I've just added you as additional QE